### ISSUE TITLE
x/website/tour: installing @latest gets older version

### ISSUE THREAD

**Issue by dmitshur** [golang/go#47236]
A reasonable way to install the latest version of the `tour` binary from its latest location is by using `go install`:

```
go install golang.org/x/website/tour@latest
```

However, doing so currently gets a slightly older version of the tour, before the x/website/tour nested module was merged into the x/website module in [CL 323897](https://golang.org/cl/323897).

<details><br>

```
$ go install golang.org/x/website/tour@latest
go: downloading golang.org/x/website v0.0.0-20210715154925-11adafea15f8
go: downloading golang.org/x/website/tour v0.0.0-20210616181959-e0d934b43647

$ go version -m $(which tour) | head -n 3
/Users/dmitshur/go/bin/tour: go1.16.6
	path	golang.org/x/website/tour
	mod	golang.org/x/website/tour	v0.0.0-20210616181959-e0d934b43647	h1:IFK+1/z6bm51XIJD3ksijCgwsZXJJxliWFiIYVEG6F8=
```

Note that module path is `golang.org/x/website/tour` and version is a month-old `v0.0.0-20210616181959-e0d934b43647`, rather than the latest commit on `master` branch:

```
$ go install golang.org/x/website/tour@HEAD
$ go version -m $(which tour) | head -n 3  
/Users/dmitshur/go/bin/tour: go1.16.6
	path	golang.org/x/website/tour
	mod	golang.org/x/website	v0.0.0-20210715154925-11adafea15f8	h1:xRFLa1Fo4NnOCH0yKaPu6H36MmmA/GJEZyTEO+HOWlI=
```

</details>

A workaround is to specify `@HEAD` or the exact version of tour one wants.

**Comment by dmitshur** [comment]
Hmm, this doesn't seem to happen in GOPROXY=direct mode, so maybe the module mirror is playing a role. I'll check again later.

**Comment by dmitshur** [comment]
I understand why this is happening and can confirm both the go command and module mirror are working as expected. Specifically, https://golang.org/ref/mod#resolve-pkg-mod specifies that:

> If one or more modules contain the requested package, the module with the longest path is used.

When the module mirror is used, the latest versions of both x/website and x/website/tour are found, both contain the `tour` package, and the x/website/tour with the its longer module path gets selected (despite being older).

I suspect we'll need to fix this by pushing a newer version of the x/website/tour module that retracts and/or deletes the `tour` package.

**Comment by bcmills** [comment]
#39007 is closely related.

**Comment by ericfrederich** [comment]
I guess this is still an issue in 2025.
I can only run a 4 year old version (2021) of the tour locally?
Is there a workaround?

This seems like it would be an important bug but no attention in 4 years?

```
# go install golang.org/x/website/tour@latest
go: downloading golang.org/x/website v0.0.0-20250219015331-21ca6346b037
# ~/go/bin/tour
2025/02/19 09:15:33 Serving content from /home/ubuntu/go/pkg/mod/golang.org/x/website/tour@v0.0.0-20210616181959-e0d934b43647
2025/02/19 09:15:33 Please open your web browser and visit http://127.0.0.1:3999
```

**Comment by dmitshur** [comment]
@ericfrederich Yes, there is a workaround, as mentioned in the original report:

> A workaround is to specify `@HEAD` or the exact version of tour one wants.

For example, `go run golang.org/x/website/tour@v0.0.0-20250219015331-21ca6346b037` runs a recent version of tour:

```
$ go run golang.org/x/website/tour@v0.0.0-20250226202246-0620964c30c6 
2025/02/26 16:55:40 A browser window should open. If not, please visit http://127.0.0.1:3999
[…]
```

That said, I agree this bug is unfortunate and would be good to fix since beginners are less likely to discover that there is a workaround.

CC @golang/tools-team via [owners](https://dev.golang.org/owners).

**Comment by dmitshur** [comment]
We've retracted all versions of the unintended x/website/tour module in [v0.1.0](https://pkg.go.dev/golang.org/x/website/tour@v0.1.0?tab=versions) (including v0.1.0 itself), so now installing or running the tour with the "@latest" version query gets the latest version from the x/website module as intended:

```
$ go install golang.org/x/website/tour@latest
go: downloading golang.org/x/website v0.0.0-20250226202246-0620964c30c6
$ go version -m $(which tour) | head -n 3    
$HOME/go/bin/tour: go1.24.0
	path	golang.org/x/website/tour
	mod	golang.org/x/website	v0.0.0-20250226202246-0620964c30c6	h1:JtlmRtOHMUWcPnv9zPCom6XYsUBjblK9HlmhF7hw7Pk=
```

Also, https://golang.org/x/website/tour now defaults to displaying documentation for the latest x/website version instead of the x/website/tour from 2021.

Closing.
